config/config.go

```diff
@@ -169,6 +175,14 @@ func newFromYaml(data []byte) (*Config, error) {
 		return nil, fmt.Errorf("Failed to parse YAML config: %v", err)
 	}
 
+	if c.HTTPAddress == "" {
+		c.HTTPAddress = net.JoinHostPort(c.Host, strconv.Itoa(c.Port))
+	}
+
+	if c.GRPCAddress == "" && c.GRPCPort > 0 {
+		c.GRPCAddress = net.JoinHostPort(c.Host, strconv.Itoa(c.GRPCPort))
+	}
```

Comment:
This is a bit tricky since `c.Port` takes on a default value for purposes of backwards compatibility (in `utils/flags/flags.go`). So at this point in the logic, `c.Port` is `8080` even when it is omitted from the command line. (Same for the gRPC counterpart.) What do you think?

Reply:
Oh, right. I guess we can add more error checking later if it's important.